### ISSUE TITLE
Cannot create invoice w/ route hints [Embedded node]

### ISSUE THREAD

**Issue by Darth-Coin** [ZeusLN/zeus#2489]
### Describe your enhancement idea

Zeus v0.9.1 on Android 12 (Lineage)
2 channels: 1 with Olympus, 1 with Blixt.

### Creating a new invoice with route hints get error "edge not found"
Steps to reproduce:
- The Blixt channel have more inbound liquidity (20x more than the invoice amount) and I want to receive in this one.
- So on the main screen I swipe the LN box and select "receive".
- Put the amount and then select "include route hints" - custom.
- Select the channel to receive
- hit "create invoice"

I get this:

![zeus-1](https://github.com/user-attachments/assets/d05c98f9-3d1c-472a-a169-493d4639ddb7)

If I do the same but selecting Olympus channel, it works perfectly fine and also the payment goes through fine.
Sending out from both channels works fine.
The node is synced perfectly fine, 

![zeus-3](https://github.com/user-attachments/assets/cd61db2a-ea5c-45e9-974c-7d4cb0f026ed)

The logs shows nothing special or other errors, only that 
[ERR] RPCS: [/lnrpc.Lightning/GetChanInfo]: edge not found

```
15:45:40.054 [INF] DISC: Obtained 1 addrs to bootstrap network with
15:45:40.225 [INF] SRVR: Established connection to: 031e649898e3e07847e4c09ce421a3144821a8951862046c5c741c252b47400588@85.208.69.17:4062
15:45:40.225 [INF] SRVR: Finalizing connection to 031e649898e3e07847e4c09ce421a3144821a8951862046c5c741c252b47400588@85.208.69.17:4062, inbound=false
15:45:40.329 [INF] PEER: Peer(031e649898e3e07847e4c09ce421a3144821a8951862046c5c741c252b47400588): Negotiated chan series queries
15:45:40.329 [INF] DISC: Creating new GossipSyncer for peer=031e649898e3e07847e4c09ce421a3144821a8951862046c5c741c252b47400588
15:45:40.329 [INF] DISC: GossipSyncer(031e649898e3e07847e4c09ce421a3144821a8951862046c5c741c252b47400588): applying gossipFilter(start=2024-10-23 15:45:40.329229786 +0000 UTC m=+37.574897546, end=2160-11-29 22:13:55.329229786 +0000 UTC)
15:45:58.611 [INF] GRPH: Initial zombie prune starting
15:45:58.611 [INF] GRPH: Examining channel graph for zombie channels
15:46:01.473 [INF] GRPH: Pruning 9 zombie channels
15:46:01.642 [INF] CHDB: Pruned unconnected node 02d368f8f44eef55b34eda6fdaf3aceba95a4386afe2988b9ec9529cc36f6323f7 from channel graph
15:46:01.644 [INF] CHDB: Pruned unconnected node 03adef1d075a73263065fa69b91bf9a9364121ec4f4e6f2676c40950e9a1e5d204 from channel graph
15:46:01.645 [INF] CHDB: Pruned unconnected node 03f0d3ecf271ae3fef1783c60662c5b3b652fc3c5c7f1da1da14f3fe2517da8635 from channel graph
15:46:01.645 [INF] CHDB: Pruned 3 unconnected nodes from the channel graph
15:48:11.085 [INF] GRPH: Processed channels=1 updates=69 nodes=0 in last 2m42.476764415s
15:49:11.085 [INF] GRPH: Processed channels=0 updates=45 nodes=0 in last 59.999310717s
15:49:13.292 [INF] RPCS: elapsed time for diameter (13) calculation: 57.561152ms
15:50:11.085 [INF] GRPH: Processed channels=0 updates=44 nodes=2 in last 59.9997228s
15:51:11.085 [INF] GRPH: Processed channels=0 updates=113 nodes=0 in last 1m0.000074415s
15:51:21.503 [INF] BTCN: Processed 1150 blocks in the last 6m0.61s (height 867027, 2024-10-23 15:51:08 +0000 UTC)
15:51:21.511 [INF] BTCN: Attempting to fetch set of un-checkpointed filters at height=867027, hash=00000000000000000001df2c27c1f42e9c1f5b7c605c24c1129a0a86efbbdbda
15:51:21.569 [INF] BTCN: Verified 1150 filter headers in the last 6m0.68s (height 867027, 2024-10-23 15:51:08 +0000 UTC)
15:51:22.199 [INF] NTFN: New block: height=867027, sha=00000000000000000001df2c27c1f42e9c1f5b7c605c24c1129a0a86efbbdbda
15:51:22.199 [INF] LNWL: Web API returning 1012 sat/kvb for min relay feerate
15:51:22.201 [INF] UTXN: Attempting to graduate height=867027: num_kids=0, num_babies=0
15:52:07.870 [ERR] RPCS: [/lnrpc.Lightning/GetChanInfo]: edge not found
15:52:11.085 [INF] GRPH: Processed channels=0 updates=65 nodes=16 in last 59.998786862s
15:52:31.365 [INF] RPCS: elapsed time for diameter (13) calculation: 84.389956ms
15:52:33.832 [ERR] LNWL: No min relay fee rate available, using default 253 sat/kw
```

**Comment by Darth-Coin** [comment]
Just FYI, I had the Olympus channel closed due to expiring time.
I opened a new one, buying from LSP.
And now is not working also on the Olympus LSP to choose the rote hints with Olympus. Again it says "edge not found".

What I tried: switched to reset graph and restart. Wait for a while, watching the logs to see when is finished to be sure I have all the graph synced.
Then tried again. Didn't worked.
2nd intent - I tried switching to Blixt graph sync, restart. Tried again, selecting the Olympus channels as route hints. Didn't worked again. Tried with Blixt channel and also didn't worked.

Is confirmed, if I choose a simple Zeus wrapped invoice is working, but received only in the Olympus channel.

**Comment by Darth-Coin** [comment]
Some more updates.
Updated to v0.9.3 beta 1
I tried also resetting the graph and compacted LND db. Also delete neutrino files.
Switched to Blixt graph option.

If I choose the rote hints "automatic" the invoice is created and paid correctly.
But is always choosing the LSP channel, ignoring all the others.

Another observation: if I choose "blinded paths" (with route hints deactivated) it return the error "could not find any routes to self to use for blinded route construction". With previous versions this option worked fine.
